ruuvi_examples/ruuvi_firmware/main.c

```diff
@@ -282,97 +296,149 @@ ret_code_t lis2dh12_int2_handler(const ruuvi_standard_message_t message)
 
-
-
-    // Setup BME280 - oversampling must be set for each used sensor.
-    bme280_set_oversampling_hum(BME280_HUMIDITY_OVERSAMPLING);
-    bme280_set_oversampling_temp(BME280_TEMPERATURE_OVERSAMPLING);
+            // oversampling must be set for each used sensor.
+    bme280_set_oversampling_hum  (BME280_HUMIDITY_OVERSAMPLING);
+    bme280_set_oversampling_temp (BME280_TEMPERATURE_OVERSAMPLING);
     bme280_set_oversampling_press(BME280_PRESSURE_OVERSAMPLING);
     bme280_set_iir(BME280_IIR);
     bme280_set_interval(BME280_DELAY);
     bme280_set_mode(BME280_MODE_NORMAL);
-    NRF_LOG_DEBUG("BME280 configuration done\r\n");
-  }
-
-  // Visually display init status. Hangs if there was an error, waits 3 seconds on success.
-  init_blink_status(err_code);
-
-  nrf_gpio_pin_set(LED_RED);  // Turn RED led off.
-  // Turn green led on to signal model +
-  // LED will be turned off in power_manage.
-  if (model_plus) { nrf_gpio_pin_clear(LED_GREEN); }
-
-  // Delay before advertising so we get valid data on first packet
-  nrf_delay_ms(MAIN_LOOP_INTERVAL_RAW + 100);
-
-  // Init ok, start watchdog with default wdt event handler (reset).
-  init_watchdog(NULL);
-
-  // start advertising
-  bluetooth_advertising_start();
-
-  // Enter main loop.
+    NRF_LOG_DEBUG("BME280 configuration done \r\n");                            // HOW TO init_status |= BME_FAILED_INIT; (_)
+  }   
+
+    // show to production QA and anyone else who's watching
+    if (init_status ) { NRF_LOG_WARNING (" -- Initalization error :  %d \r\n", init_status);
+                        for ( int16_t i=0; i<13; i++){ RED_LED_ON; nrf_delay_ms(100u); RED_LED_OFF; nrf_delay_ms(200u); }
+                        nrf_delay_ms(1000u);  }
+ 
+
+  if (model_plus) GREEN_LED_ON;    // will be turned off in power_manage.
+
+  nrf_delay_ms(MAIN_LOOP_INTERVAL_RAW + 100); // Delay before advertising so we get valid data on first packet
+
+  bluetooth_advertising_start(); NRF_LOG_INFO("Advertising start  \r\n");           // can this fail ?
+
+  if (init_status) {     // Report on status of initalization 
+   url_buffer[9]=init_status>>8;url_buffer[10]=init_status & 0xFF;       //  URL ruu.vi/#xx
+   bluetooth_set_eddystone_url(url_buffer, sizeof(url_buffer) ); nrf_delay_ms(500u); // keep sending this for a while
+//  raspberry pi:  
+//  export bdaddr='zz yy xx ww vv uu' #  MAC address least significant byte FIRST
+//  hcidump --raw |g  --line-buffered --after-context=2 $bdaddr
+//  04 3E 2B 02 01 00 01 C2 24 68 3C 10 C7 1F 02 01 06 03 03 AA
+//                       zz yy xx ww vv uu
+//  FE 17 16 AA FE 10 F6 03 72 75 75 2E 76 69 2F 23 10 00 00 00
+//                          r  u  u  .  v  i  /  #  ^^^^^--init_status example BATTERY_FAILED_INIT
+
+  // like format 03  too
+   data_buffer[0]   =   0x13 ;                              // NOT SENSOR_TAG_DATA_FORMAT 03;  (but could be intreperted as such)

```

Comment:
What's in this data?

Reply:
This data is just an error packet (format 13, bad luck), similar to a format 03 packet, filled with invalid values, including the cute "I DID BAD" which turns out to be easy to spot, and finally the init_status and the vbat.

Reply:
why 13 and not 0? 13 may be useful for an "actual" format later on